### DISCUSSION TITLE
Weird behavior of form field :: Value not coming

### DISCUSSION THREAD

**Discussion by shamseerahammedm** [react-hook-form/react-hook-form#9930] [General]
my form is not printing phone number, but it comes if u uncomment `<h3>` tag in this example sandbox. With out giving `h3` tag it wont come ( which I think something related to re rendering )

`watch()` functions returns the entire form values, phone number comes there but wont show up in page

Sandbox - https://codesandbox.io/s/inspiring-danny-yqjtww?file=/src/App.js

What i am doing wrong, help on this is appreciated, thanks

**Comment by leapful** [comment]
@shamseerahammedm 

Please always remember to put prop `key` with unique value for element that can be re-used on re-rendering.

In your example, flag `isNTBKeyPerson` is used to render field `phone` and other fields, otherwise, it will render `registeredAddresss` but your implementation is using same `<div>` HTML element type therefore React will try to re-use the real DOM div element when the boolean flag is changed and causes wrong form field value.

Working demo: https://codesandbox.io/s/quizzical-perlman-z4krdt?file=/src/App.js
